operator/internal/manifests/internal/config/loki-config.yaml

```diff
@@ -161,6 +161,121 @@ schema_config:
 ruler:
   enable_api: true
   enable_sharding: true
+  {{ with .Ruler.EvaluationInterval }}
+  evaluation_interval: {{ . }}
+  {{ end }}
+  {{ with .Ruler.PollInterval }}
+  poll_interval: {{ . }}
+  {{ end }}
+  {{ with .Ruler.AlertManager }}
+  external_url: {{ .ExternalURL }}
+  {{ with .ExternalLabels }}
+  external_labels:
+    {{ range $name, $value := . }}
+    {{ $name }}: {{ $value }}
+    {{ end }}
+  {{ end}}
+  alertmanager_url: {{ .Hosts }}
+  {{ if .EnableV2 }}
+  enable_alertmanager_v2: {{ .EnableV2 }}
+  {{ end }}
+  {{ if .EnableDiscovery }}
+  enable_alertmanager_discovery: {{ .EnableDiscovery }}
+  alertmanager_refresh_interval: {{ .RefreshInterval }}
+  {{ end }}
+  {{ if .QueueCapacity }}
+  notification_queue_capacity: {{ .QueueCapacity }}
+  {{ end }}
+  {{ if .Timeout }}
+  notification_timeout: {{ .Timeout }}
+  {{ end }}
+  {{ if .ForOutageTolerance }}
+  for_outage_tolerance: {{ .ForOutageTolerance }}
+  {{ end }}
+  {{ if .ForGracePeriod }}
+  for_grace_period: {{ .ForGracePeriod }}
+  {{ end }}
+  {{ if .ResendDelay }}
+  resend_delay: {{ .ResendDelay }}
+  {{ end }}
+  {{ end}}
+  {{ with .Ruler.RemoteWrite }}
+  {{ if .Enabled }}
+  remote_write:
+    enabled: {{ .Enabled }}
+    config_refresh_period: {{ .RefreshPeriod }}
+    client:
+      {{  with .Client }}
+      name: {{ .Name }}
+      url: {{ .URL }}
+      timeout: {{ .RemoteTimeout }}
+      proxy_url: {{ .ProxyURL }}
+      follow_redirects: {{ .FollowRedirects }}
+      headers:
+      {{ range $key, $val := .Headers }}
+        {{ $key }}: {{ $val }}
+      {{ end }}
+      {{ if and .BasicAuthUsername .BasicAuthPassword }}
```

Comment:
Do we want to support only authenticating with username (empty password)? I know that the Github API used this as a replacement for the bearer-auth in the past, so this might be a pattern that someone might need.